src/webgl/p5.RendererGL.js

```diff
@@ -1233,72 +1233,90 @@ p5.RendererGL.prototype.getTexture = function(img) {
 p5.RendererGL.prototype._setStrokeUniforms = function(strokeShader) {
   strokeShader.bindShader();
 
-  // set the uniform values
-  strokeShader.setUniform('uMaterialColor', this.curStrokeColor);
-  strokeShader.setUniform('uStrokeWeight', this.curStrokeWeight);
+  if (strokeShader !== this.userStrokeShader) {
+    // set the uniform values
+    strokeShader.setUniform('uMaterialColor', this.curStrokeColor);
+    strokeShader.setUniform('uStrokeWeight', this.curStrokeWeight);
+    strokeShader.setUniform('uViewport', this._viewport);
+    if (this._curCamera.cameraType === 'default') {
+      // strokes scale up as they approach camera, default
+      strokeShader.setUniform('uPerspective', 1);
+    } else {
+      // strokes have uniform scale regardless of distance from camera
+      strokeShader.setUniform('uPerspective', 0);
+    }
+  }
 };
 
 p5.RendererGL.prototype._setFillUniforms = function(fillShader) {
   fillShader.bindShader();
 
-  // TODO: optimize
-  fillShader.setUniform('uMaterialColor', this.curFillColor);
-  fillShader.setUniform('isTexture', !!this._tex);
-  if (this._tex) {
-    fillShader.setUniform('uSampler', this._tex);
-  }
-  fillShader.setUniform('uTint', this._tint);
+  if (fillShader !== this.userFillShader) {
+    // TODO: optimize
+    fillShader.setUniform('uMaterialColor', this.curFillColor);
+    fillShader.setUniform('isTexture', !!this._tex);
+    if (this._tex) {
+      fillShader.setUniform('uSampler', this._tex);
```

Comment:
Wouldn't this prevent people from using the uSampler uniform? For instance, at the moment, you can use uSampler as a uniform in your shader and have p5 send you whatever texture was bound with the texture() function. I think we wouldn't want to take that functionality away.

Reply:
Just based off of the code, I think if a user has set `texture(someTexture)`, then `this._tex` will be set, and the `uSampler` uniform will still get set for you here. A test to confirm that would be great though